framework/src/base/LinearSystemContributionObject.C

```diff
@@ -55,3 +55,31 @@ LinearSystemContributionObject::LinearSystemContributionObject(const InputParame
     _mesh(_subproblem.mesh())
 {
 }
+
+void
+LinearSystemContributionObject::linkObjectsForContribution(const std::set<TagID> & vector_tags,
```

Comment:
yeah not a great name.

gatherPointersToVectorsAndMatrices is more accurate

Seems weird to intersect with getVectorTags too
the user of this API passed them in the parameter, and they could get ignored

Reply:
Yes, a kernel can contribute to multiple tags (let's say rhs and extra_tag_1). When you call the assembly loop with specific tags (only extra_tag_1), you want to make sure only the objects only contribute to that vector.